src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,556 @@
+        normalize_targets=normalize_targets[benchmark],
+        hidden_space=hidden_space[benchmark],
+        with_cost=False,
+        config=config,
+    )
+
+    net = get_network(model_objective, X_train.shape[1])
+
+    if task_id is None:
+        task_id = rng.randint(Y.shape[0])
+
+    h = task_features_mean[task_id] + task_features_std[task_id] * rng.randn(
+        hidden_space[benchmark]
+    )
+
+    return net, h
+
+
+def load_task_network(
+    checkpoint_file: Union[str, Path]
+) -> Tuple[nn.Module, np.ndarray]:
+    with open(checkpoint_file, "rb") as f:
+        state = pickle.load(f)
+
+    network = get_architecture[state["benchmark"]](state["size"])
+    network.load_state_dict(state["network"])
+    h = state["h"]
+
+    return network, h
+
+
+def compute_identity(size: int = 16, **sample) -> str:
+    """Compute a unique hash out of a dictionary
+
+    Parameters
+    ----------
+    size: int
+        size of the unique hash
+
+    **sample:
+        Dictionary to compute the hash from
+
+    """
+    sample_hash = hashlib.sha256()
+
+    for k, v in sorted(sample.items()):
+        sample_hash.update(k.encode("utf8"))
+
+        if isinstance(v, dict):
+            sample_hash.update(compute_identity(size, **v).encode("utf8"))
+        else:
+            sample_hash.update(str(v).encode("utf8"))
+
+    return sample_hash.hexdigest()[:size]
+
+
+def save_task_network(
+    checkpoint_file: Union[str, Path], benchmark: str, network: nn.Module, h: np.ndarray
+) -> None:
+    checkpoint_file = Path(checkpoint_file)
+    state = dict(
+        benchmark=benchmark,
+        network=network.state_dict(),
+        size=list(network.parameters())[0].size()[1],
+        h=h.tolist(),
+    )
+
+    tmp_file = checkpoint_file.with_suffix(".tmp")
+    with open(tmp_file, "wb") as file:
+        pickle.dump(state, file, protocol=pickle.DEFAULT_PROTOCOL)
+    tmp_file.rename(checkpoint_file)
+
+
+class ProfetTask(Task):
+    def __init__(
+        self,
+        benchmark: str,
+        task_id: Optional[int]=0,
+        input_dir: Union[Path, str] = None,
+        rng: np.random.RandomState = None,
+        seed: int = 1,
+        checkpoint_dir: Union[Path, str] = None,
+        max_trials: int = 100,
+        train_config: MetaModelTrainingConfig = None,
+        **kwargs,
+    ):
+        if rng is None:
+            rng = np.random.RandomState(seed)
+        task_id = task_id or 0
+        super().__init__(
+            task_id=task_id,
+            rng=rng,
+            seed=seed,
+            max_trials=max_trials,
+            fixed_dims=kwargs,
+        )
+        self.benchmark = benchmark
+        if not input_dir:
+            input_dir = Path(os.environ.get("DATA_DIR")) / "profet"
+        else:
+            input_dir = Path(input_dir)
+
+        if checkpoint_dir is None:
+            checkpoint_dir = input_dir / "checkpoints"
+        # TODO: Save the train config in the `configuration` of the task.
```

Comment:
Would `configuration` of the task be a configuration object (data-class) like MetaModelTrainingConfig?

Reply:
Hmm interesting idea. Although since nothing else is doing that at the moment, I'll instead just use a regular dict like everywhere else.